sql/tests: TestRandomSyntaxGeneration failed

[(sql/tests).TestRandomSyntaxGeneration failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2836513&tab=buildLog) on [master@41f921d1401d1df4a16f98be758702281551b2eb](https://github.com/cockroachdb/cockroach/commits/41f921d1401d1df4a16f98be758702281551b2eb):

```
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +Wraps: (2)
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +Wraps: (3) attached stack trace
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  -- stack trace:
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog.(*validationErrorAccumulator).decorate
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/validate.go:300
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog.(*validationErrorAccumulator).Report
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/validate.go:244
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | [...repeated from below...]
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +Wraps: (4) schema "ident" (260)
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +Wraps: (5) assertion failure
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +Wraps: (6) attached stack trace
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  -- stack trace:
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog/schemadesc.(*Immutable).ValidateCrossReferences
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/schemadesc/schema_desc.go:190
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog.Validate.func2
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/validate.go:65
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog.(*validationErrorAccumulator).validateDescriptorsAtLevel
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/validate.go:255
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog.Validate
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/validate.go:60
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog.ValidateSelfAndCrossReferences
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/validate.go:147
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog/catalogkv.GetAllDescriptors
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/catalogkv/catalogkv.go:300
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.(*Collection).GetAllDescriptors
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/descs/collection.go:1720
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | github.com/cockroachdb/cockroach/pkg/sql.forEachTableDescWithTableLookupInternal
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:2053
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | github.com/cockroachdb/cockroach/pkg/sql.forEachTableDescWithTableLookup
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:2007
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | github.com/cockroachdb/cockroach/pkg/sql.makeAllRelationsVirtualTableWithDescriptorIDIndex.func1
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pg_catalog.go:1032
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | github.com/cockroachdb/cockroach/pkg/sql.(*virtualDefEntry).getPlanInfo.func1.1
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_schema.go:529
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | github.com/cockroachdb/cockroach/pkg/sql.setupGenerator.func3
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_table.go:121
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:351
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | runtime.goexit
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1374
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +Wraps: (7) not present in parent database [233] schemas mapping
E210401 00:51:45.228102 2707747 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55256,hostssl,user=root] 263 +Error types: (1) *colexecerror.notInternalError (2) *colexecerror.notInternalError (3) *withstack.withStack (4) *errutil.withPrefix (5) *assert.withAssertionFailure (6) *withstack.withStack (7) *errutil.leafError
    rsg_test.go:752: Crash detected: 
        SHOW INDEX FROM DATABASE GEOMETRYCOLLECTIONM WITH COMMENT
        
        Stack trace:
    rsg_test.go:761: 1085778 executions, 12967 successful
    rsg_test.go:229: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRandomSyntaxGeneration238956369
--- FAIL: TestRandomSyntaxGeneration (300.62s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestRandomSyntaxGeneration PKG=./pkg/sql/tests TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRandomSyntaxGeneration.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/tests).TestRandomSyntaxGeneration failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2846454&tab=buildLog) on [master@6e5dfc46eab59f7fd6e77ee5dbab5bac715330de](https://github.com/cockroachdb/cockroach/commits/6e5dfc46eab59f7fd6e77ee5dbab5bac715330de):

```
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:50
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql.startExec
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:502
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Start
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:121
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Columnarizer).Init
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/columnarizer.go:156
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql/colexec.(*InvariantsChecker).Init
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/invariants_checker.go:52
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:91
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Materializer).Start
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/materializer.go:273
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBase).Run
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:773
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:382
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:359
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:967
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1040
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:912
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:656
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:120
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1527
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1529
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1450
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:484
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:626
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +Wraps: (6) not present in parent database [139] schemas mapping
E210404 05:24:49.703694 2823604 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:55128,hostssl,user=root] 8343 +Error types: (1) *colexecerror.notInternalError (2) *withstack.withStack (3) *errutil.withPrefix (4) *assert.withAssertionFailure (5) *withstack.withStack (6) *errutil.leafError
    rsg_test.go:752: Crash detected: 
        REASSIGN OWNED BY STRING , 'string' , 'string' TO CURRENT_USER
        
        Stack trace:
    rsg_test.go:761: 1113484 executions, 12630 successful
E210404 05:24:49.715660 2823864 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 8344  making txn commit explicit failed for "unnamed" meta={id=e93d0d30 pri=100.00000000 epo=0 ts=1617513889.711947910,0 min=1617513889.711947910,0 seq=2} lock=true stat=STAGING rts=1617513889.711947910,0 wto=false gul=1617513890.211947910,0 ifw=1: result is ambiguous (server shutdown)
    rsg_test.go:229: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRandomSyntaxGeneration027715606
--- FAIL: TestRandomSyntaxGeneration (300.18s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestRandomSyntaxGeneration PKG=./pkg/sql/tests TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #63054 sql/tests: TestRandomSyntaxGeneration failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-21.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-21.1)

- #63016 sql/tests: TestRandomSyntaxGeneration failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.2)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRandomSyntaxGeneration.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/tests).TestRandomSyntaxGeneration failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2851133&tab=buildLog) on [master@f676f07152b8d0ca203831bb9e75d9b7d3ed5b79](https://github.com/cockroachdb/cockroach/commits/f676f07152b8d0ca203831bb9e75d9b7d3ed5b79):

```
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +(1)
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +Wraps: (2) attached stack trace
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  -- stack trace:
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog.(*validationErrorAccumulator).decorate
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/validate.go:303
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog.(*validationErrorAccumulator).Report
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/validate.go:244
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | [...repeated from below...]
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +Wraps: (3) schema "ident" (92)
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +Wraps: (4) assertion failure
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +Wraps: (5) attached stack trace
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  -- stack trace:
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog/schemadesc.(*Immutable).ValidateCrossReferences
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/schemadesc/schema_desc.go:190
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog.Validate.func2
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/validate.go:65
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog.(*validationErrorAccumulator).validateDescriptorsAtLevel
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/validate.go:255
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog.Validate
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/validate.go:60
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog.ValidateSelfAndCrossReferences
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/validate.go:147
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog/catalogkv.GetAllDescriptors
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/catalogkv/catalogkv.go:300
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backupresolver.LoadAllDescs.func1
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backupresolver/targets.go:496
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn.func1
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:772
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:828
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:771
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backupresolver.LoadAllDescs
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backupresolver/targets.go:492
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.backupPlanHook.func3
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/backupccl/backup_planning.go:841
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/planhook.go:145
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | runtime.goexit
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1374
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +Wraps: (6) not present in parent database [82] schemas mapping
E210406 05:23:28.637492 2481076 sql/sqltelemetry/report.go:57  [n1,client=127.0.0.1:58450,hostssl,user=root] 10075 +Error types: (1) *colexecerror.notInternalError (2) *withstack.withStack (3) *errutil.withPrefix (4) *assert.withAssertionFailure (5) *withstack.withStack (6) *errutil.leafError
    rsg_test.go:752: Crash detected: 
        BACKUP INTO 'string' IN ISERROR WITH ENCRYPTION_PASSPHRASE = INNER , REVISION_HISTORY , DETACHED
        
        Stack trace:
    rsg_test.go:761: 990713 executions, 10828 successful
    rsg_test.go:229: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRandomSyntaxGeneration718063735
--- FAIL: TestRandomSyntaxGeneration (300.20s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestRandomSyntaxGeneration PKG=./pkg/sql/tests TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #63054 sql/tests: TestRandomSyntaxGeneration failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-21.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-21.1)

- #63016 sql/tests: TestRandomSyntaxGeneration failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.2)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRandomSyntaxGeneration.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/drop_schema.go
+++ b/pkg/sql/drop_schema.go
@@ -174,8 +174,8 @@ func (p *planner) dropSchemaImpl(
 	if parentDB.Schemas == nil {
 		parentDB.Schemas = make(map[string]descpb.DatabaseDescriptor_SchemaInfo)
 	}
-	parentDB.Schemas[parentDB.Name] = descpb.DatabaseDescriptor_SchemaInfo{
-		ID:      parentDB.ID,
+	parentDB.Schemas[sc.GetName()] = descpb.DatabaseDescriptor_SchemaInfo{
+		ID:      sc.GetID(),
 		Dropped: true,
 	}
 	// Mark the descriptor as dropped.

--- a/pkg/sql/logictest/testdata/logic_test/schema
+++ b/pkg/sql/logictest/testdata/logic_test/schema
@@ -611,3 +611,36 @@ query TT
 SELECT schema_name, table_name FROM [SHOW TABLES FROM for_show.sc1]
 ----
 sc1    t1
+
+# Regression test for #62920. The bug that motivated this test would populate
+# the schema entry in the database with the database's name rather than the
+# schemas.
+subtest schema_and_database_with_same_name
+
+statement ok
+CREATE DATABASE samename
+
+statement ok
+USE samename
+
+statement ok
+CREATE SCHEMA foo;
+CREATE SCHEMA bar
+
+statement ok
+DROP SCHEMA foo
+
+statement ok
+CREATE SCHEMA samename
+
+statement ok
+DROP SCHEMA bar
+
+statement ok
+CREATE TABLE samename.samename.t (i INT PRIMARY KEY)
+
+statement ok
+SHOW TABLES
+
+statement ok
+DROP DATABASE samename CASCADE;

